src/signature.c

```diff
@@ -121,6 +122,44 @@ int git_signature_dup(git_signature **dest, const git_signature *source)
 	return 0;
 }
 
+int git_signature_with_mailmap(
+	git_signature **dest,
+	const git_signature *source,
+	const git_mailmap *mailmap)
+{
+	git_signature *signature = NULL;
+	const char *name = NULL;
+	const char *email = NULL;
+
+	if (source == NULL)
+		goto on_error;
+
+	git_mailmap_resolve(&name, &email, mailmap, source->name, source->email);
```

Comment:
Shouldn't you check for errors here? While `GIT_ENOTFOUND` is okay, you might for exampleg et allocation errors

Reply:
git_mailmap_resolve returns `void`.